core/constant_inc.php

```diff
@@ -591,8 +592,11 @@
 define( 'DB_FIELD_SIZE_PASSWORD', 64 );
 define( 'DB_FIELD_SIZE_API_TOKEN_NAME', 128 );
 
-# Maximum size for the user's password when storing it as a hash
-define( 'PASSWORD_MAX_SIZE_BEFORE_HASH', 1024 );
+# Maximum size for the user's password when storing it as a hash.
+# The CRYPT_BLOWFISH algorithm used by password_hash() will silently truncate
+# passwords to 72 characters, so there is no point in allowing longer passwords.
+# @see http://php.net/function.crypt
+define( 'PASSWORD_MAX_SIZE_BEFORE_HASH', 72 );
```

Comment:
I would rather keep the limit as before.  For example, if an auth plugin that validates passwords supports a longer length (e.g. LDAP), then we shouldn't block it.  Let's not make decisions based on one specific algorithm.